daft/dataframe/dataframe.py

```diff
@@ -567,7 +555,51 @@ def write_iceberg(self, table: "pyiceberg.table.Table", mode: str = "append") ->
             rows.append(data_file.record_count)
             size.append(data_file.file_size_in_bytes)
 
-        merge.commit()
+        if parse(pyiceberg.__version__) >= parse("0.7.0"):
+            from pyiceberg.table import ALWAYS_TRUE, PropertyUtil, TableProperties
+
+            tx = table.transaction()
+
+            if unsupported_partitions := [
+                field for field in tx.table_metadata.spec().fields if not field.transform.supports_pyarrow_transform
+            ]:
+                raise ValueError(
+                    f"Not all partition types are supported for writes. Following partitions cannot be written using pyarrow: {unsupported_partitions}."
+                )
```

Comment:
Wait, so partitioned write **are** supported? But certain transforms arent'?

Reply:
If that's the case, do we have tests yet for partitioned writes on tables with "simple" partition transforms (e.g. identity)?

Reply:
Ah good catch. Missed this when I ported the logic over from pyiceberg. What we should really be doing is checking if the table is partitioned and error if it is

Reply:
Ok checked the code and we already raise an error on a partitioned table and the method to do that is the same between pyiceberg versions. I removed this check since it won't do anything right now